src/svd/bindings.py

```diff
@@ -0,0 +1,988 @@
+    # Write access is permitted. Read operations have an undefined result.
+    WRITE_ONLY = "write-only"
+    # Read and write accesses are permitted.
+    READ_WRITE = "read-write"
+    # Only the first write after reset has an effect. Read operations have an undefined results.
+    WRITE_ONCE = "writeOnce"
+    # Only the first write after reset has an effect. Read access is permitted.
+    READ_WRITE_ONCE = "read-writeOnce"
+
+
+AccessElement = make_enum_wrapper(Access)
+
+
+@enum.unique
+class ReadAction(CaseInsensitiveStrEnum):
+    """Side effect following a read operation of a given register or field."""
+
+    # The register/field is set to zero following a read operation.
+    CLEAR = "clear"
+    # The register/field is set to ones following a read operation.
+    SET = "set"
+    # The register/field is modified by a read operation.
+    MODIFY = "modify"
+    # A dependent resource is modified by a read operation.
+    MODIFY_EXTERNAL = "modifyExternal"
+
+
+ReadActionElement = make_enum_wrapper(ReadAction)
+
+
+@enum.unique
+class Endian(CaseInsensitiveStrEnum):
+    """Processor endianness."""
+
+    # Little endian
+    LITTLE = "little"
+    # Big endian
+    BIG = "big"
+    # Endianness is configurable for the device, taking effect on the next reset.
+    SELECTABLE = "selectable"
+    # Neither big nor little endian
+    OTHER = "other"
+
+
+EndianElement = make_enum_wrapper(Endian)
+
+
+@enum.unique
+class SauAccess(CaseInsensitiveStrEnum):
+    """SAU region access type"""
+
+    # Non-secure accessible
+    NON_SECURE = "n"
+    # Secure callable
+    SECURE_CALLABLE = "c"
+
+
+SauAccessElement = make_enum_wrapper(SauAccess)
+
+
+@enum.unique
+class AddressBlockUsage(CaseInsensitiveStrEnum):
+    """Defined usage type of a peripheral address block."""
+
+    REGISTER = "registers"
+    BUFFERS = "buffers"
+    RESERVED = "reserved"
+
+
+AddressBlockUsageElement = make_enum_wrapper(AddressBlockUsage)
+
+
+@enum.unique
+class Protection(CaseInsensitiveStrEnum):
+    """Security privilege required to access an address region"""
+
+    # Secure permission required for access
+    SECURE = "s"
+    # Non-secure or secure permission required for access
+    NON_SECURE = "n"
+    # Privileged permission required for access
+    PRIVILEGED = "p"
+
+
+ProtectionElement = make_enum_wrapper(Protection)
+
+
+@enum.unique
+class EnumUsage(CaseInsensitiveStrEnum):
+    """Usage of an enumerated value."""
+
+    # The value is relevant for read operations.
+    READ = "read"
+    # The value is relevant for write operations.
+    WRITE = "write"
+    # The value is relevant for read and write operations.
+    READ_WRITE = "read-write"
+
+
+EnumUsageElement = make_enum_wrapper(EnumUsage)
+
+
+@enum.unique
+class WriteAction(CaseInsensitiveStrEnum):
+    """Side effect following a write operation of a given register or field."""
+
+    # Bits written to one are set to zero in the register/field.
+    ONE_TO_CLEAR = "oneToClear"
```

Comment:
Are all of these enumerations from the SVD spec? If so can we make note of it in the class docstring?